### PR TITLE
OCaml Tour: -New sections- Introduction and Before We Begin. Added REPL definition and double semicolon use

### LINKED ISSUE #2324: (Docs) Shorten Introduction on "Tour of OCaml" and Mention Double Semicolon
The introduction on https://ocaml.org/docs/tour-of-ocaml needs to
1. be shortened
2. say (briefly!) what a REPL or toplevel is, and
3. mention why `;;` needs to be used.

The corresponding document is https://github.com/ocaml/ocaml.org/blob/main/data/tutorials/getting-started/1_01_a_tour_of_ocaml.md

### PR DESCRIPTION
**Pull request by Alfredo-Carlon**
Resolves #2324 

Hi, in this I explicitly mark the _Introduction_ and _Prerequisites_ (Before We Begin) sections. I think it is easier to read, but I'm not sure if the Introduction got shorter :-)
In the prerequisites I also added REPL's definition and on the first example why `;;` is required.

### CODE REVIEW

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
This first part introduces OCaml's basic features: values, expressions, lists, functions, pattern matching and more.
In the second part, we'll see how to create our first OCaml project in the [Your First OCaml Program](/docs/your-first-program) tutorial.
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: Yes, we assume programming (implicitly imperative/oo) but not functional programming

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
We recommend you walk through this tutorial by trying out its code samples (and your variations) in UTop (Universal Toplevel), an OCaml [REPL](https://en.wikipedia.org/wiki/Read-eval-print_loop). It allows interacting directly with OCaml. Phrases are read, evaluated and results print on the screen. Launch UTop by using the `utop` command.
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Thanks, @Alfredo-Carlon I think this going in the right direction. Here are suggestions trying to push even further.

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (Alfredo-Carlon) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: I think it is best to only state the aim of the tutorial. This way the writing has a positive tone

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (Alfredo-Carlon) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: I'm not entirely sure about this sentences shift. 
The "We recommend..." sentence is a nice closing sentence with the meaning: we gave you this info and this is how we think  you should use it for the upcoming part.

But now that you suggest it, I'm thinking about the possibility of getting rid of REPL altogether.
During the tour you don't need to know what a REPL is in the general sense (like it is now in the doc) but, you do need to know UTop reads, evals and print what you type. 
I'll push another commit trying to mix these past two suggestions.

Comment (cuihtlauac) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: > But now that you suggest it, I'm thinking about the possibility of getting
> rid of REPL altogether.
> During the tour you don't need to know what a REPL is in the general sense
> (like it is now in the doc) but, you do need to know UTop reads, evals and
> print what you type.

I agree.

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md:50]: Should we already state that double semicolon is not going to be needed when compiling files?

**Review by christinerose — APPROVED** [review]

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (Alfredo-Carlon) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md:50]: Probably not because I don't think we should ask the reader to keep that in mind through the first part of the tour. It adds little value to this first part. But it is important to state it somewhere, I have seen FP programmers (unfamiliar with OCaml) using `;;` in `.ml` files so it would be valuable to address it, but I think the best place to do it is in the second part.

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md:39]:
```suggestion
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md:53]:
```suggestion
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
To walk through the basics of OCaml by trying out different elements in an interactive manner, we'll use UTop. You can run UTop by using the `utop` command. Exit it by pressing `Ctrl+D`.

UTop (Universal Toplevel) allows interacting directly with OCaml by reading and evaluating OCaml phrases like expressions or value definitions and printing the result on the screen.
```

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (cuihtlauac) [inline data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]:
```suggestion
In this tutorial, we'll have a look at the most commonly used language features.
```
Let's make this more boring.

**Review by cuihtlauac — DISMISSED** [review]
@Alfredo-Carlon What is your opinion on thoses suggestions?

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (Alfredo-Carlon) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: I liked it fun. My initial thought was the bits about "hop on..." should not be just at the summary, but also in the text itself. But ok, let's make it a bit more boring. 
I think It might be a good idea to work more on the introduction after we go thru the text (in another PR)

File: data/tutorials/getting-started/1_01_a_tour_of_ocaml.md
Comment (Alfredo-Carlon) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: @cuihtlauac 
Yes we need to mention how to exit UTop :-D
I don't think the shift of sentences helps us. 
This is my reasoning:
1st Sentence: Why we'll use something call UTop
2nd Sentence: What is that thing that we will use.
3rd Sentence: How you can start using it.
If we shift the sentences  I feel like it breaks a bit the dynamic.
Does this make sense?

Comment (cuihtlauac) [reply data/tutorials/getting-started/1_01_a_tour_of_ocaml.md]: The order you propose is fine for me. Lets just have 
```
You can run UTop by using the `utop` command. Exit it by pressing `Ctrl+D`.
```

**Review by sabine — APPROVED** [review]
Thanks @Alfredo-Carlon, after shortening this some, I'm happy to merge this after @christinerose did line editing! :heart:

**Review by christinerose — APPROVED** [review]
Good to go! Thanks for waiting to merge. 🙂

**Review by cuihtlauac — APPROVED** [review]
Thanks @Alfredo-Carlon. This looks good to me too

**Pull request merged**
